// ngx-typed-js/esm2020/lib/ngx-typed-js.module.mjs
import { NgModule } from '@angular/core';
import { NgxTypedJsComponent } from './ngx-typed-js.component';
import * as i0 from "@angular/core";
export class NgxTypedJsModule {
}
NgxTypedJsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.1.1", ngImport: i0, type: NgxTypedJsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgxTypedJsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.1.1", ngImport: i0, type: NgxTypedJsModule, declarations: [NgxTypedJsComponent], exports: [NgxTypedJsComponent] });
NgxTypedJsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.1.1", ngImport: i0, type: NgxTypedJsModule, imports: [[]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.1.1", ngImport: i0, type: NgxTypedJsModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [NgxTypedJsComponent],
                    imports: [],
                    exports: [NgxTypedJsComponent]
                }]
        }] });
